BuildHelpers/Public/Get-BuildVariables.ps1

```diff
@@ -75,7 +75,7 @@ function Get-BuildVariables {
     $Path = ( Resolve-Path $Path ).Path
     $Environment = Get-Item ENV:
     if(!$PSboundParameters.ContainsKey('GitPath')) {
-        $GitPath = (Get-Command $GitPath -ErrorAction SilentlyContinue)[0].Path
+        $GitPath = (Get-ChildItem ($env:PATH -split ';') -filter git.exe -ErrorAction SilentlyContinue | Select-Object -First 1).Fullname
```

Comment:
I need to fix the fact that git.exe is hard coded here. I will either need to change the `-filter` to something like `-filter "*$GitPath*" or change the param default to git.exe

Reply:
I think I've changed my mind on this. If the user provides a git path they should be providing a full path to the git executable. If they provide anything at all this variable is given directly to the Process Start info Filename Parameter anyway, so if they do something like provide an input that it can't use, it will already fail.

If the user does not provide input to the $GitPath parameter, we already know conclusively that we want to search for git.exe. We aren't searching for git.com or git.bat, or anything like that.

This says to me that it is safe to simply leave the $GitPath parameter optional, and there is no need to provide a default value.

Reply:
Hiyo!  @gaelcolas has an old PR in one of my repos for a similar thing, looking to resolve just `git` (this should still pick up `git.exe`, but buys us compatibility with systems that don't use `.exe`).  Might be worth looking into something like this here?  (This is a `pre-skimming-your-whole-commit-and-my-old-code` comment : P)